web/src/components/input/CustomSearchInput.js

```diff
@@ -64,8 +64,10 @@ const CustomSearchInput = ({label, exclude, formData, onChange, fullWidth}) => {
       }
     }
     else {
-      // Programmatic change or 'clear'
-      if(textValue !== value) {
+      // Programmatic change or 'clear' || run check on init case
+      // it is used to handle case where the species no longer exist but
+      // used in supersededby.
+      if(textValue !== value || searchTerm === null) {
         // Update if diff to avoid looping
         setSearchTerm(value);
```

Comment:
Can you please clarify "clear" in this context? If it's when `bird` is removed before `eagle` and expecting `superseded by` field value when editing `eagle` to be emptied, then it doesn't work.
![image](https://github.com/aodn/nrmn-application/assets/26201635/78049b1a-686d-4eaf-a0a6-32558c9db0c0)

and also by reading :

> it is used to handle case where the species no longer

without previous context knowledge, I assume line 72 would be reached only when the description above happens. However, it seems when root species still exist, the line still be reached. e.g `sample` is parent of `pigeon`

![image](https://github.com/aodn/nrmn-application/assets/26201635/8c4376bc-ef47-436d-84a9-a6143e0ec8f7)

and when `bird` is removed while it was parent of `eagle`

![image](https://github.com/aodn/nrmn-application/assets/26201635/57876a47-ccfb-45ba-ba34-14d047acc824)

other than that, the main issue which was a blank page is gone.

Reply:
> Can you please clarify "clear" in this context? If it's when `bird` is removed before `eagle` and expecting `superseded by` field value when editing `eagle` to be emptied, then it doesn't work. ![image](https://user-images.githubusercontent.com/26201635/252570119-78049b1a-686d-4eaf-a0a6-32558c9db0c0.png)
> 
> and also by reading :
> 
> > it is used to handle case where the species no longer
> 
> without previous context knowledge, I assume line 72 would be reached only when the description above happens. However, it seems when root species still exist, the line still be reached. e.g `sample` is parent of `pigeon`
> 
> ![image](https://user-images.githubusercontent.com/26201635/252571850-8c4376bc-ef47-436d-84a9-a6143e0ec8f7.png)
> 
> and when `bird` is removed while it was parent of `eagle`
> 
> ![image](https://user-images.githubusercontent.com/26201635/252572166-57876a47-ccfb-45ba-ba34-14d047acc824.png)
> 
> other than that, the main issue which was a blank page is gone.

Refer to this https://mui.com/material-ui/api/autocomplete oninputchange, i think it means text box clear.

The code reach is after the value set in the field, it will trigger a lookup in the database and verify if the mentioned species is a valid in db. So if you delete the species, it will prompt for error.

Reply:
cool thanks @utas-raymondng for explaining